sdk/go/docker/docker.go

```diff
@@ -225,7 +225,7 @@ func buildImage(ctx *pulumi.Context, imageName string, build *DockerBuild,
 	// If the container build specified build stages to cache, build each in turn.
 	var stages []string
 	if build.CacheFrom != nil && build.CacheFrom.Stages != nil {
-		for _, stage := range stages {
+		for _, stage := range build.CacheFrom.Stages {
```

Comment:
Linting correctly pointed out this loop would never do anything as originally written.